proto/substrait/plan.proto

```diff
@@ -80,3 +84,12 @@ message Version {
   // created the plan.
   string producer = 5;
 }
+
+// Represents a binding for a dynamic parameter.
+message ParameterBinding {
+  // The parameter anchor that identifies the dynamic parameter reference.
+  uint32 parameter_anchor = 1;
+
+  // The literal value assigned to the parameter at runtime.
```

Comment:
Maybe add a comment that the type of this is only valid if it matches the type declared in the DynamicParameter